VineyardInternal/Person.swift

```diff
@@ -7,22 +7,35 @@
 
 import Foundation
 
-class Person: Identifiable {
-    var id: String
+struct Person: Identifiable {
+    let id: UUID = .init()
     var name: String
-    var age: Int
-
-    init(id: String = UUID().uuidString, name: String, age: Int) {
-        self.id = id
-        self.name = name
-        self.age = age
+    var groups: [Group] = []
+    var allProgress: [Progress] = []
+    var email: String
+    var badges: [Badge] = []
+    
+    mutating func addGroup(_ group: Group) {
+        self.groups.append(group)
     }
     
-    func toDictionary() -> [String: Any] {
-        return [
-            "id": self.id,
-            "name": self.name,
-            "age": self.age
-        ]
+    static var samples: [Person] {
+        let group1 = Group.samples[0]
+        let group2 = Group.samples[1]
+        
+        let progress1 = Progress.samples[0]
+        let progress2 = Progress.samples[1]
+        
+        let badge1 = Badge.samples[0]
+        let badge2 = Badge.samples[1]
+        
+        let andrew = Person(name: "Andrew", groups: [group1], allProgress: [progress1], email: "a@gmail.com")
```

Comment:
both Andrew and yash should hold both progresses. since they are both part of a group with those resolutions.